Multiple ADI devices failing to build in CI CMSIS portability tests

**Describe the bug**
Currently 16 combinations of platforms and samples are failing in weekly CI in ADI platforms 

* samples/subsys/portability/cmsis_rtos_v2/*

x

* max32655fthr/max32655/m4
* max32655evkit/max32655/m4
* max32680evkit
* max32690evkit

Due to re definitions of the TRUE/FALSE macros in the HAL and CMSIS

**To Reproduce**
Steps to reproduce the behavior:
1. twister -T samples/subsys/portability/cmsis_rtos_v2/philosophers/ -s sample.portability.cmsis_rtos_v2.philosopher  -p max32655evkit/max32655/m4

**Expected behavior**
No build warnings which are promoted to errors in CI

**Impact**
16 failures in weekly CI

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/runs/25996829280
```
In file included from /__w/zephyr/zephyr/subsys/portability/cmsis_rtos_v2/thread.c:13:
/__w/zephyr/zephyr/subsys/portability/cmsis_rtos_v2/wrapper.h:13: error: "TRUE" redefined [-Werror]
   13 | #define TRUE    1
      | 
In file included from /__w/zephyr/modules/hal/adi/MAX/./Include/wrap_max32xxx.h:37,
                 from /__w/zephyr/zephyr/soc/adi/max32/soc.h:15,
                 from /__w/zephyr/zephyr/modules/cmsis/./cmsis_core_m.h:24,
                 from /__w/zephyr/zephyr/modules/cmsis/./cmsis_core.h:10,
                 from /__w/zephyr/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24,
                 from /__w/zephyr/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
                 from /__w/zephyr/zephyr/include/zephyr/arch/arm/arch.h:33,
                 from /__w/zephyr/zephyr/include/zephyr/arch/cpu.h:19,
                 from /__w/zephyr/zephyr/include/zephyr/kernel_includes.h:36,
                 from /__w/zephyr/zephyr/include/zephyr/kernel.h:17,
                 from /__w/zephyr/zephyr/subsys/portability/cmsis_rtos_v2/thread.c:7:
/__w/zephyr/modules/hal/adi/MAX/Libraries/CMSIS/Device/Maxim/MAX32655/Include/max32655.h:37: note: this is the location of the previous definition
   37 | #define TRUE (1)
      | 
/__w/zephyr/zephyr/subsys/portability/cmsis_rtos_v2/wrapper.h:14: error: "FALSE" redefined [-Werror]
   14 | #define FALSE   0
      | 
/__w/zephyr/modules/hal/adi/MAX/Libraries/CMSIS/Device/Maxim/MAX32655/Include/max32655.h:33: note: this is the location of the previous definition
   33 | #define FALSE (0)
      | 
```

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Toolchain (e.g Zephyr SDK, ...): 0.16.8
 - Commit SHA or Version used: Main as of now b46fb464cbaaad8d094f75487a3502a11126d025

--- a/subsys/portability/cmsis_rtos_v2/wrapper.h
+++ b/subsys/portability/cmsis_rtos_v2/wrapper.h
@@ -10,8 +10,12 @@
 #include <zephyr/kernel.h>
 #include <cmsis_os2.h>
 
+#ifndef TRUE
 #define TRUE    1
+#endif
+#ifndef FALSE
 #define FALSE   0
+#endif
 
 struct cv2_thread {
 	sys_dnode_t node;

